components/brave_wallet_ui/page/screens/send/send/send.tsx

```diff
@@ -344,6 +361,43 @@ export const Send = (props: Props) => {
     }
   }, [selectedTokensNetwork])
 
+  React.useEffect(() => {
+    // check if the user has selected an asset
+    if (!contractAddress || !accountAddress || !chainId || selectedSendAsset) return
+
+    const asset = sendAssetOptions.find((option) =>
+      tokenId
+        ? option.contractAddress.toLowerCase() ===
+            contractAddress.toLowerCase() && option.tokenId === tokenId
+        : option.contractAddress.toLowerCase() === contractAddress.toLowerCase()
+    )
+    const account = accounts.find(
+      (account) => account.address === accountAddress
+    )
+
+    if (!asset || !account) return
+    
+    dispatch(WalletActions.selectAccount(account))
+    const _setNetwork = async () =>
+      await setNetwork({
+        chainId: chainId,
+        coin: asset.coin
+      })
+    _setNetwork()
```

Comment:
How about just not await-ing the original function?
```suggestion
    setNetwork({
      chainId: chainId,
      coin: asset.coin
    })
```

Reply:
I think you need to do `await setNetwork` to fix the SonarCloud error.

Reply:
@josheleonard I changed this to use Promises. As @onyb mentions, without `await` SonaCloud fails